WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListActivity.java

```diff
@@ -264,4 +304,55 @@ private ReaderPostListFragment getListFragment() {
         }
         return ((ReaderPostListFragment) fragment);
     }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (requestCode == RequestCodes.EDIT_POST && resultCode == Activity.RESULT_OK
+            && data != null && !isFinishing()) {
+            int localId = data.getIntExtra(EditPostActivity.EXTRA_POST_LOCAL_ID, 0);
+            final SiteModel site = (SiteModel) data.getSerializableExtra(WordPress.SITE);
+            final PostModel post = mPostStore.getPostByLocalPostId(localId);
+
+            if (EditPostActivity.checkToRestart(data)) {
+                ActivityLauncher.editPostOrPageForResult(data, ReaderPostListActivity.this, site,
+                        data.getIntExtra(EditPostActivity.EXTRA_POST_LOCAL_ID, 0));
+                // a restart will happen so, no need to continue here
+                return;
+            }
+
+            if (site != null && post != null) {
+                mUploadUtilsWrapper.handleEditPostResultSnackbars(
+                        this,
+                        findViewById(R.id.coordinator),
+                        data,
+                        post,
+                        site,
+                        mUploadActionUseCase.getUploadAction(post),
+                        new View.OnClickListener() {
+                            @Override
+                            public void onClick(View v) {
+                                UploadUtils.publishPost(ReaderPostListActivity.this, post, site, mDispatcher);
+                            }
+                        });
+            }
```

Comment:
Yes. It is actually the 3rd place that repeats this logic since it was already in WPMainActivity.
It the logic should be delegated to a common component or use the Activity Result APIs introduced in AndroidX https://developer.android.com/training/basics/intents/result
I was thinking of proceeding with the PRs and open another issue for this since it could also touch the main activity.